catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -550,12 +560,98 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          WorkspaceMetacardImpl workspace = workspaceService.getWorkspaceMetacard(id);
+
+          String[] queryIds = workspace.getQueries().toArray(new String[0]);
+
+          if (queryIds.length > 0) {
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
+          }
+
           catalogFramework.delete(new DeleteRequestImpl(id));
+
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    get(
```

Comment:
✏️ All the `/workspace` and `/query` bits would be good to see go into their own `RouteGroup`s. That way each could truly have a single responsibility and this class, `MetacardApplication`, would only serve as a place to delegate `path`s from.

Reply:
...this class is pretty massive as it stands, > 1k lines, and does all sorts of thangs